tests/src/python/test_qgsserver_wms_getmap.py

```diff
@@ -1561,7 +1561,7 @@ def test_wms_getmap_filter_ogc_v2(self):
 
         # without namespace (only with prefix)
         filter = (
-            "<fes:Filter>"
+            "<fes:Filter xmlns:fes='http://www.opengis.org/fes'>"
```

Comment:
The comment above explicitly says "without namespace". Isn't this breaking what the test is designed to check?

Reply:
Indeed, I missed the comment! Without a namespace, it makes the XML not valid so I do not understand what is expected here or how it should work.

I investigated further and the error throws by the test is from https://github.com/benoitdm-oslandia/QGIS/blob/79bb4814dfd689d60eeaf15f28455e0321c27b1c/src/server/services/wms/qgswmsrenderer.cpp#L3563. What I do not understand (again) is how this can works with Qt6 as the `setContent` function has not the same parameters as Qt5?! Can you enlighten me?

Reply:
Ok it is now a deprecated function and hidden in the documentation. I will fix

Reply:
This shouldn't have been marked resolved -- it's still outstanding

Reply:
wrong click!

Reply:
I have made a (ugly) fix

Reply:
As stated [here](https://doc.qt.io/qt-6/xml-changes-qt6.html#qdom-and-qdomdocument)

> Undeclared namespace prefixes are no longer allowed.

That's why it failed with Qt6. I think a way to nicely fix these would be to use the [setContent](https://doc.qt.io/qt-6/qdomdocument.html#setContent-2) that takes a QXmlStreamReader and call [addExtraNamespaceDeclaration](https://doc.qt.io/qt-6/qxmlstreamreader.html#addExtraNamespaceDeclaration) on it before so we are sure its gonna work when parsing the filter

Reply:
That approach sounds reasonable to me!